### PR TITLE
Shoatman/dev/20

### PR DESCRIPTION
**Pull request by shoatman**

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/adal/internal/util/JsonExtensions.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.Iterator;
+import java.util.Map;
+
+
+/**
+ * Helper methods related to JSON
+ */
+public class JsonExtensions {
+
+    public static Map<String, String> extractJsonObjectIntoMap(final String jsonString)
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/util/JsonExtensions.java]: I know this is just being 'pulled-in' from ADAL but there's something about it's functionality I thought I would at least call your attention to: it 'flattens' JSON objects, meaning nested objects need to call this method multiple times to access inner-fields.

For instance:
```json
{
	"a": "Aardvark",
	"b": "Banana",
	"c": "Cardamom",
	"d": {
		"1": "Cyclops",
		"2": "Biclops",
		"3": "Triclops",
		"4": "Quadclops?"
	}
}
```

Returns a `Map<String, String` containing four entries:
```
"a" : "Aardvark"
"b" : "Banana"
"c" : "Cardamom"
"d" : "{"1":"Cyclops","2":"Biclops","3":"Triclops","4":"Quadclops?"}"
```

I went digging around in the code to test this before I posted this comment - I also found _three_ implementations of it peppered around the code:

1. [IdToken](https://github.com/AzureAD/azure-activedirectory-library-for-android/blob/dev/adal/src/main/java/com/microsoft/aad/adal/IdToken.java#L161-L170)
2. [OAuth2](https://github.com/AzureAD/azure-activedirectory-library-for-android/blob/dev/adal/src/main/java/com/microsoft/aad/adal/Oauth2.java#L342-L352)
3. [SignInActivity](https://github.com/AzureAD/azure-activedirectory-library-for-android/blob/dev/automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/SignInActivity.java#L208-L217)

File: common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/IDToken.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.Collections;
+import java.util.Map;
+
+import com.microsoft.identity.common.adal.internal.util.JsonExtensions;
+import com.microsoft.identity.common.adal.internal.util.StringExtensions;
+
+/**
+ * Represents the ID Token returned from the oAuth Authorization Server / OpenID Connect Provider
+ */
+public class IDToken {
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/IDToken.java]: Nit: The Android style guide suggests treating [acronyms as words](https://source.android.com/setup/code-style#treat-acronyms-as-words). Can we call this class `IdToken`?

Comment (shoatman) [reply common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/IDToken.java]: I'll correct this in my next PR... Thanks Brian

File: common/src/main/java/com/microsoft/identity/common/adal/internal/util/JsonExtensions.java (hunk 1 above)
Comment (shoatman) [reply common/src/main/java/com/microsoft/identity/common/adal/internal/util/JsonExtensions.java]: Yep.  Saw that.  In this case it's ok... I'll file another issue around improving JSON handling in general.  This method avoids deserializing to a specific type... which makes sense in some cases and not in others.  You could argue that this isn't the correct method for dealing with ClientInfo for example...

**Pull request merged**